qutebrowser/browser/webengine/webview.py

```diff
@@ -261,3 +264,30 @@ def acceptNavigationRequest(self,
                                                  is_main_frame=is_main_frame)
         self.navigation_request.emit(navigation)
         return navigation.accepted
+
+    def chooseFiles(self, mode, old_files, accepted_mimetypes):
+        """Override chooseFiles to (optionally) invoke custom file uploader."""
+        handler = config.val.fileselect.handler
+        if handler == "default":
+            return super().chooseFiles(mode, old_files, accepted_mimetypes)
+
+        with tempfile.NamedTemporaryFile(delete=False) as tmpfile:
+            filename = tmpfile.name
+
+        if mode == QWebEnginePage.FileSelectOpen:
+            command = config.val.fileselect.single_file.command
+        else:
+            command = config.val.fileselect.multiple_files.command
+
+        proc = guiprocess.GUIProcess(what='choose-file')
+        proc.start(command[0],
+                   [arg.replace('{}', filename) for arg in command[1:]])
+
+        loop = QEventLoop()
+        proc.finished.connect(lambda code, status: loop.exit())
```

Comment:
```suggestion
        proc.finished.connect(lambda _code, _status: loop.exit())
```

to mark them as unused.